src/assets/CurrencyRateController.ts

```diff
@@ -195,30 +195,47 @@ export class CurrencyRateController extends BaseController<
       pendingCurrentCurrency,
       pendingNativeCurrency,
     } = this.state;
+
+    const conversionDate: number = Date.now() / 1000;
+    let conversionRate: number | null = null;
+    let usdConversionRate: number | null = null;
     try {
-      const {
-        conversionDate,
-        conversionRate,
-        usdConversionRate,
-      } = await this.fetchExchangeRate(
-        pendingCurrentCurrency || currentCurrency,
-        pendingNativeCurrency || nativeCurrency,
-        this.includeUsdRate,
-      );
-      this.update(() => {
-        return {
-          conversionDate,
-          conversionRate,
-          currentCurrency: pendingCurrentCurrency || currentCurrency,
-          nativeCurrency: pendingNativeCurrency || nativeCurrency,
-          pendingCurrentCurrency: null,
-          pendingNativeCurrency: null,
-          usdConversionRate,
-        };
-      });
+      if (
+        (pendingCurrentCurrency || currentCurrency) &&
```

Comment:
We should really use the same pattern here and in the `fetchExchangeRate` invocation as below, to ensure we're using the exact same `nativeCurrency` and `currentCurrency` as we're setting below.

With the addition of the empty string guard here for `pendingNativeCurrency` that is.

Reply:
If we use the null coalescer in the if we should also check if `nativeCurrency` is an empty string since that would pass through. Is there any reason to not just change the arguments in the invocation to use the null coalescing but keep the || checks in the if? Like this:
```
if (
        (pendingCurrentCurrency || currentCurrency) &&
        (pendingNativeCurrency || nativeCurrency) &&
        pendingNativeCurrency !== ''
      ) {
        ({ conversionRate, usdConversionRate } = await this.fetchExchangeRate(
          pendingCurrentCurrency ?? currentCurrency,
          pendingNativeCurrency ?? nativeCurrency,
          this.includeUsdRate,
        ));
```

Reply:
Ah, I see what you mean. Hmm.

My concern was that we want it to be clear at a glance that the same `nativeCurrency` and `currentyCurrency` is being considered in the condition, passed to `fetchExchangeRate`, and later saved in state. If either of these three things get out-of-sync it could lead to some pretty difficult-to-debug problems.

Maybe this is best accomplished by using a variable? e.g.

```
const currentCurency = pendingCurrentCurrency ?? stateCurrentCurrency;
const nativeCurrency = pendingNativeCurrency ?? stateNativeCurrency;

if (currentCurrency && nativeCurrency) {
  ...
}
```

Then these can be used in the `fetchExchangeRate` invocation and the state change as well. That would ensure we're always using the same value, whatever it gets resolved to.